schemas/event-schema.json

```diff
@@ -0,0 +1,77 @@
+{
+	"$schema": "http://json-schema.org/draft-07/schema#",
+	"$id": "http://schema.esamarathon.com/mq/event-schema",
+	"title": "European Speedrunner Assembly RabbitMQ event schema",
+	"definitions": {
+		"nonEmptyString": {
+			"type": "string",
+			"minLength": 1
+		},
+		"timestampObject": {
+			"type": "object",
+			"required": [
+				"iso",
+				"unix"
+			],
+			"properties": {
+				"iso": {
+					"type": "string",
+					"description": "Timestamp representation in the ISO 8601 format",
+					"format": "date-time",
+					"examples": [
+						"2019-09-03T19:55:18.430Z"
+					]
+				},
+				"unix": {
+					"type": "number",
+					"description": "Timestamp representation in seconds since the Unix epoch, including a fractional millisecond part",
+					"examples": [
+						1567540518.430
+					]
+				}
+			},
+			"additionalProperties": false
+		}
+	},
+	"properties": {
+		"$meta": {
+			"type": "object",
+			"required": [
+				"vHost",
+				"name",
+				"exchange",
+				"routingKey",
+				"ownerTool"
+			],
+			"properties": {
+				"vHost": {
+					"$ref": "#/definitions/nonEmptyString",
+					"description": "VHost this event is emitted to"
+				},
+				"name": {
+					"$ref": "#/definitions/nonEmptyString",
+					"description": "Name of this message type"
+				},
+				"exchange": {
+					"$ref": "#/definitions/nonEmptyString",
+					"description": "RabbitMQ exchange name this event is emitted to"
+				},
+				"routingKey": {
+					"$ref": "#/definitions/nonEmptyString",
+					"description": "RabbitMQ routing key (topic) this event is emitted to"
```

Comment:
Given how a lot of our routingKeys are pretty complex and use various forms of shell variable/expansion syntax are more than just a single routingKey, this should probably be expanded to reflect that.

Following how the definitions look today, they are using <var_name> for variable elements that are filled with data, and {a,b,c} syntax to define elements that have multiple possible values.
Additionally, it might be good to define how multiple independent keys might look. I'd say just comma-separate them.

Reply:
What the parts of the routing key mean is described in the README at least.

Reply:
I added a reference to the README file in the description of `$meta.routingKey`. I feel like describing the entire format in the description here would be a little unnecessary, so hopefully this will give people enough of a hint for how to use this. Technically a regex could be cooked up for this too, but I feel like that's a little overkill.